python/pyarrow/tests/test_table.py

```diff
@@ -239,23 +239,65 @@ def test_concat_tables():
 
 
 def test_table_pandas():
-    data = [
-        pa.array(range(5)),
-        pa.array([-10, -5, 0, 5, 10])
-    ]
-    table = pa.Table.from_arrays(data, names=('a', 'b'))
-
-    # TODO: Use this part once from_pandas is implemented
-    # data = {'a': range(5), 'b': [-10, -5, 0, 5, 10]}
-    # df = pd.DataFrame(data)
-    # pa.Table.from_pandas(df)
+    data = {'a': range(5), 'b': [-10, -5, 0, 5, 10]}
+    df = pd.DataFrame(data)
+    table = pa.Table.from_pandas(df)
 
     df = table.to_pandas()
     assert set(df.columns) == set(('a', 'b'))
     assert df.shape == (5, 2)
     assert df.loc[0, 'b'] == -10
 
 
+def test_table_pandas_with_schema():
+    data = OrderedDict({
+        'a': range(5),
+        'b': [-10, -5, 0, 5, 10],
+        'c': np.array([-10, -5, 0, 5, 10], dtype=np.int32)
+    })
+    df = pd.DataFrame(data)
+
+    schema = pa.schema([
+        pa.field('a', pa.int64()),
+        pa.field('b', pa.int64()),
+        pa.field('c', pa.int32())
+    ])
+    table = pa.Table.from_pandas(df, schema=schema)
```

Comment:
Thanks for the quick response! Yes, you're right. I'll move the tests to `python/pyarrow/tests/test_convert_pandas.py` as suggested by @xhochy and will remove the duplicates.